src/main/java/io/apicurio/datamodels/core/validation/rules/other/SecurityRequirementUniquenessValidationRule.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.datamodels.core.validation.rules.other;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.apicurio.datamodels.core.Constants;
+import io.apicurio.datamodels.core.models.common.SecurityRequirement;
+import io.apicurio.datamodels.core.validation.ValidationRule;
+import io.apicurio.datamodels.core.validation.ValidationRuleMetaData;
+import io.apicurio.datamodels.openapi.v2.models.Oas20Operation;
+
+/**
+ * @author cfoskin@redhat.com
+ */
+public class SecurityRequirementUniquenessValidationRule extends ValidationRule {
+
+    /**
+     * Constructor.
+     * 
+     * @param ruleInfo
+     */
+    public SecurityRequirementUniquenessValidationRule(ValidationRuleMetaData ruleInfo) {
+        super(ruleInfo);
+    }
+
+    /**
+     * @param currentSecurityRequirement
+     * @param otherSecurityRequirement
+     * @return
+     */
+    private boolean isEqualTo(SecurityRequirement currentSecurityRequirement,
+            SecurityRequirement otherSecurityRequirement) {
+       
+        // first check the names of the security requirements we need to compare
+        List<String> currentSecurityRequirementNames = currentSecurityRequirement
+                .getSecurityRequirementNames();
+        List<String> otherSecurityRequirementNames = otherSecurityRequirement.getSecurityRequirementNames();
+        // sort the lists before checking if they are equal
+        Collections.sort(currentSecurityRequirementNames);
+        Collections.sort(otherSecurityRequirementNames);
+        
+        // check if they are equal
+        if (currentSecurityRequirementNames.equals(otherSecurityRequirementNames)) {
+            int matchesFound = 0;
+            // if they have a list of the same names, we need to check the scopes for each security
+            // requirement name against
+            // the scopes for all of the other security requirements - who have the same name (to keep
+            // context).
+            for (int i = 0; i < currentSecurityRequirementNames.size(); i++) {
```

Comment:
You should be able to use the enhanced for loop syntax in Java:  https://www.programiz.com/java-programming/enhanced-for-loop